src/pages/settings/PreferencesPage.js

```diff
@@ -38,42 +48,74 @@ const priorityModes = {
-                    <RNPickerSelect
-                        onValueChange={
-                            mode => NameValuePair.set(CONST.NVP.PRIORITY_MODE, mode, ONYXKEYS.PRIORITY_MODE)
-                        }
-                        items={Object.values(priorityModes)}
-                        style={styles.picker}
-                        useNativeAndroidPickerStyle={false}
-                        placeholder={{}}
-                        value={priorityMode}
-                        Icon={() => <Icon src={DownArrow} />}
-                    />
+class PreferencesPage extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isNewsStatusEnabled: !(this.props.user.expensifyNewsStatus === false),
+        };
+        this.toggleNewsStatus = this.toggleNewsStatus.bind(this);
+    }
+
+    toggleNewsStatus(toggle) {
+        this.setState({isNewsStatusEnabled: toggle});
+        setExpensifyNewsStatus(toggle);
+    }
+
+    render() {
+        return (
+            <>
+                <HeaderGap />
+                <HeaderWithCloseButton
+                    title="Preferences"
+                    shouldShowBackButton
+                    onBackButtonPress={() => redirect(ROUTES.SETTINGS)}
+                    onCloseButtonPress={() => redirect(ROUTES.HOME)}
+                />
+                <View style={styles.pageWrapper}>
+                    <View style={[styles.settingsPageBody, styles.mb6]}>
+                        <Text style={[styles.formLabel]} numberOfLines={1}>Notifications</Text>
+                        <View style={[styles.flexRow, styles.mb5, styles.justifyContentBetween]}>
+                            <View style={styles.flex4}>
+                                <Text>
+                                    Receive relevant feature updates and Expensify news
+                                </Text>
+                            </View>
+                            <View style={[styles.flex1, styles.alignItemsEnd]}>
+                                <ToggleSwitch
+                                    isOn={this.state.isNewsStatusEnabled}
+                                    onColor={colors.green}
+                                    onToggle={this.toggleNewsStatus}
+                                />
+                            </View>
+                        </View>
+                        <Text style={[styles.formLabel]} numberOfLines={1}>
+                            Priority Mode
+                        </Text>
+                        <View style={[styles.mb2]}>
+                            {/* empty object in placeholder below to prevent default */}
+                            {/* placeholder from appearing as a selection option. */}
+                            <RNPickerSelect
+                                onValueChange={
+                                    mode => NameValuePair.set(CONST.NVP.PRIORITY_MODE, mode, ONYXKEYS.PRIORITY_MODE)
```

Comment:
NAB (Not a blocker) but per our Philosophy https://github.com/Expensify/Expensify.cash/#Philosophy only Actions should manage onyx data so I think this should probably be added into a method on the User action `updatePriorityMode` which then calls to set the NVP. I think this is the place we'd be setting this NVP though so I'm not 100% sure if it's needed so don't consider it a blocker for now.

Reply:
This was actually existing, but since I changed the component to a class it appears as new. I don't mind setting up the method if needed.

Reply:
@bondydaa I think we should keep this as is. It seems superfluous to have separate methods for getting and setting every NVP that we modify in E.cash. Additionally, `NameValuePair` is housed in `actions` so I think it conforms enough to our current philosophy.

Reply:
Ya I missed that NVP was an action, my bad.